src/App.jsx

```diff
@@ -47,20 +51,38 @@ function App() {
       </div>
       <div className="flex w-screen">
         <NavBar status={showNavbar ? "block" : "hidden"} />
-        <div className="border min-w-10 sm:w-3/12 w-screen sm:static relative z-0">
+        <div className="flex w-screen border min-w-10 sm:static relative z-0">
           <div className="overflow-y-auto sm:min-w-64 max-h-screen">
             {emailData.map((email) => (
               <EmailItem
                 name={email.name}
                 date={formatDate(email.date_created)}
                 subject={email.subject}
                 body={email.body}
+                read={email.read}
                 id={email.id}
                 key={email.id}
-                read={email.read}
+                setCurrentEmail={setCurrentEmail}
+                setShowCloseButton={setShowCloseButton}
+                setSelectedEmail={setSelectedEmail}
+                selectedEmail={selectedEmail}
               />
             ))}
           </div>
+          <InboxDetails
+            name={currentEmail.name}
+            date={
+              currentEmail.date_created
+                ? formatDate(currentEmail.date_created)
+                : ""
+            }
+            email={currentEmail.email}
+            subject={currentEmail.subject}
+            body={currentEmail.body}
```

Comment:
Why do you pass all of this in as props? Isn't there a much easier way to get the data you need into the component?

Consider how much code you would need to change to display an additional field of an email. You would need to find and change code in `EmailItem` *and* `InboxDetails`. If you passed the `currentEmail` or even just the `id` through then `InboxDetails` would be the only place where you pick data out of that object and decide how to display it.